### ISSUE TITLE
Error without validation data

### ISSUE THREAD

**Issue by flxzimmer** [runopti/stg#8]
Running without validation data, for example: 

```
from stg import STG
from sklearn.datasets import load_breast_cancer
from sklearn.model_selection import train_test_split
import numpy as np

X, y = load_breast_cancer(return_X_y=True)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

model = STG("cpu", input_dim=X_train.shape[1], output_dim=len(np.unique(y_train)))
model.fit(X_train, y_train, nr_epochs=10)
```

I run into an error; it seems that in in `self.validate(val_data_loader, self.metric, meters)` it wants to iterate over a `val_data_loader` that is `None`.
